blob/azureblob/azureblob.go

```diff
@@ -223,6 +236,10 @@ type AccountKey string
 // https://docs.microsoft.com/en-us/azure/storage/common/storage-dotnet-shared-access-signature-part-1
 type SASToken string
 
+// CloudEnvironment is an Azure Cloud Environment domain name to target (i.e. AzureCloud, AzureUSGovernment, AzureChinaCloud).
+// It is read from the AZURE_BLOB_DOMAIN environment variable.
+type CloudEnvironment string
```

Comment:
Naming: it's a bit weird to have the type and variable called `Environment` but the docstring and env variable called `Domain`. How about using `CloudEnvironmentDomain` everywhere? (e.g., `AZURE_BLOB_ENVIRONMENT_DOMAIN`, `CloudEnvironmentDomain` type, etc.).